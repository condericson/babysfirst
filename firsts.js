const express = require('express');
const mongoose = require('mongoose');
const cloudinary = require('cloudinary');

const router = express.Router();

cloudinary.config({
  cloud_name: 'babysfirst',
  api_key: '149535471223848',
  api_secret: '1WnNvoTn2z6nee3iGpnP1R8f8uA',
});

const { Firsts } = require('./models/firstsModel');

router.get('/firsts', (req, res) => {
  Firsts.find((err, first) => {
    if (err) {
      res.status(500).json({ message: 'Error!' });
    }
    return res.status(201).json(first);
  });
});

router.post('/firsts', (req, res) => {
  let image = '';
  cloudinary.uploader.upload('req.body.image', (result) => {
    console.log(result);
    image = result;
  });
  Firsts.create({
    date: req.body.date,
    content: req.body.content,
    image,
    userId: req.body.userId,
  }, (err, first) => {
    if (err) {
      console.log(err);
      res.status(500).json({ message: 'Error with post' });
    }
    res.status(201).json(first);
  });
});

router.put('/:id', (req, res) => {
  const _id = mongoose.Types.ObjectId(req.params.id);
  Firsts.findOneAndUpdate({
    _id,
  },
    {
      $set: {
        date: req.body.date,
        content: req.body.content,
        image: req.body.image,
        userId: req.body.userId,
      },
    },
    {
      new: true,
    },
    (err, first) => {
      if (err || !first) {
        console.error('Could not update first', req.body.name);
        mongoose.disconnect();
        if (err) {
          return res.status(500).json({
            message: 'Internal Server Error',
          });
        }
      }
      console.log('Updated first on ', first.date);
      res.status(201).json(first);
    });
});

router.delete('/:id', (req, res) => {
  const _id = mongoose.Types.ObjectId(req.params.id);
  Firsts.remove({
    _id,
  }, (err, first) => {
    if (err || !first) {
      console.error('Could not delete first on ', req.body.date);
      mongoose.disconnect();
      return;
    }
    console.log('Deleted first', first.result);
    res.status(201).json(first);
  });
});

module.exports = router;
