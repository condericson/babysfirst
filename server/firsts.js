/* eslint-disable no-console */

const { Router } = require('express');
const cloudinary = require('cloudinary');

require('./config/cloudinary');
const Firsts = require('./models/firstsModel');

const router = new Router();

router.get('/', async (req, res) => {
  try {
    const first = await Firsts.find({}).sort({ date: -1 });
    res.status(201).json(first);
  } catch (e) {
    res.status(500).json({ message: 'Error!' });
  }
});

router.get('/:id', async (req, res) => {
  console.log(req.query.offset);
  const skipNumber = parseInt(req.query.offset, 10);
  try {
    const firsts = await Firsts.find({ userId: req.params.id }).sort({ date: -1 }).skip(skipNumber).limit(5);
    res.status(201).json(firsts);
  } catch (e) {
    res.status(500).json({ message: e.message });
  }
});

router.post('/', async (req, res) => {
  const { date, content, userId } = req.body;
  let image = '';
  try {
    if (req.body.image) {
      image = await cloudinary.uploader.upload(req.body.image, (result) => { console.log(result); }, { width: 1000, height: 1000, crop: 'limit' });
      if (image.state === 'rejected') {
        return console.log('Invalid image pathway');
      }
      image = image.url;
    }
    const first = await Firsts.create({ date, content, userId, image });
    return res.status(201).json(first);
  } catch (e) {
    return res.status(500).json({ message: 'Error with post' });
  }
});

// update content
router.put('/:id', async (req, res) => {
  const { content } = req.body;
  try {
    const first = await Firsts.findByIdAndUpdate(req.params.id, { content });
    return res.status(201).json(first);
  } catch (e) {
    return res.status(500).json({ message: 'Error with update' });
  }
});

router.patch('/:id', async (req, res) => {
  try {
    const first = await Firsts.findByIdAndUpdate(req.params.id, req.body);
    return res.status(200).json(first);
  } catch (e) {
    return res.status(500).json({ message: 'Internal Server Error' });
  }
});

router.delete('/:id', (req, res) => {
  console.log('ID', req.params.id);
  Firsts.findByIdAndRemove(req.params.id, (err, first) => {
    if (err || !first) {
      console.error('Could not delete first on ', req.body.date);
      return;
    }
    console.log('Deleted first', first.result);
    cloudinary.uploader.destroy(req.params.id, (result) => { console.log(result); });
    return res.status(201).json(first);
  });
});

module.exports = router;
