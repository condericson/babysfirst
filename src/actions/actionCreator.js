// add first
export function addMemory(userId, date, content, image) {
  console.log('dispatching add first');
  return {
    type: 'ADD_FIRST',
    userId,
    date,
    content,
    image,
  };
}

// // remove comment
// export function removeComment(postId, i) {
//   return {
//     type: 'REMOVE_COMMENT',
//     i,
//     postId
//   }
// }
