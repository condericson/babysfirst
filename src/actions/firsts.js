import { browserHistory } from 'react-router';
import axios from 'axios';
// import { Firsts } from '../utils/api';

export const GET_FIRSTS = 'GET_FIRSTS';
export const GET_FIRSTS_SUCCESS = 'GET_FIRSTS_SUCCESS';
export const GET_FIRSTS_ERROR = 'GET_FIRSTS_ERROR';

function getFirstsSuccess(firsts) {
  return {
    type: GET_FIRSTS_SUCCESS,
    firsts,
  };
}

function getFirstsError(error) {
  return {
    type: GET_FIRSTS_ERROR,
    error,
  };
}

export function getFirsts(userId) {
  return async dispatch => {
    dispatch({ type: GET_FIRSTS });
    try {
      const { data } = await axios.get(`/firsts/${userId}`);
      console.log('Data from axios get firsts', data);
      dispatch(getFirstsSuccess(data));
    } catch (e) {
      return dispatch(getFirstsError(e));
    }
    return browserHistory.push('/timeline');
  };
}

export const ADD_FIRST = 'ADD_FIRST';
export const ADD_FIRST_SUCCESS = 'ADD_FIRST_SUCCESS';
export const ADD_FIRST_ERROR = 'ADD_FIRST_ERROR';

function addFirstsSuccess(firsts) {
  return {
    type: ADD_FIRST_SUCCESS,
    firsts,
  };
}

function addFirstsError(error) {
  return {
    type: ADD_FIRST_ERROR,
    error,
  };
}

export function addFirsts(first) {
  return async dispatch => {
    dispatch({ type: ADD_FIRST });
    console.log('REACHED HERE');
    try {
      const { data } = await axios.post('/firsts', first);
      console.log('Data from axios first post', data);
      dispatch(addFirstsSuccess(data));
    } catch (e) {
      return dispatch(addFirstsError(e));
    }
    return browserHistory.push('/timeline');
  };
}
