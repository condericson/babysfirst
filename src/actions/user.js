import { browserHistory } from 'react-router';

import { User } from '../utils/api';

export const LOGIN = 'LOGIN';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOGIN_ERROR = 'LOGIN_ERROR';

export function loginSuccess(user) {
  return {
    type: LOGIN_SUCCESS,
    user,
  };
}

export function loginError(error) {
  return {
    type: LOGIN_ERROR,
    error,
  };
}

export function login(user) {
  return async dispatch => {
    dispatch({ type: LOGIN });
    try {
      const response = await User.login(user);
      dispatch(loginSuccess(response));
      return browserHistory.push('/timeline');
    } catch (e) {
      return dispatch(loginError(e));
    }
  };
}

export const LOGOUT = 'LOGOUT';

export function logout() {
  return {
    type: LOGOUT,
  };
}

export const SIGNUP = 'SIGNUP';
export const SIGNUP_SUCCESS = 'SIGNUP_SUCCESS';
export const SIGNUP_ERROR = 'SIGNUP_ERROR';

export function signupSuccess(user) {
  return {
    type: SIGNUP_SUCCESS,
    user,
  };
}

function signupError(error) {
  return {
    type: SIGNUP_ERROR,
    error,
  };
}

export function signup(user) {
  return async dispatch => {
    dispatch({ type: SIGNUP });
    try {
      await dispatch(signupSuccess(await User.signup(user)));
      return browserHistory.push('/timeline');
    } catch (e) {
      return dispatch(signupError(e));
    }
  };
}
