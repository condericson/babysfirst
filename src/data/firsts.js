/* eslint-disable */

const firsts = [
  {
    userId: '12345',
    date: {
      years: 1,
      months: 5,
    },
    content: '0 Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
    image: 'https://s-media-cache-ak0.pinimg.com/564x/a0/71/6c/a0716ce6fc9792f3a0ec2ae3cf67d256.jpg',
    id: 'awhjsdlaiwdaw',
  },
  {
    userId: '12345',
    date: {
      years: 1,
      months: 5,
    },
    content: '1 Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
    image: 'https://s-media-cache-ak0.pinimg.com/564x/a0/71/6c/a0716ce6fc9792f3a0ec2ae3cf67d256.jpg',
    id: 'alsijhqoilwjda',
  },
  {
    userId: '12345',
    date: {
      years: 1,
      months: 8,
    },
    content: '3 Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
    image: 'https://s-media-cache-ak0.pinimg.com/564x/a0/71/6c/a0716ce6fc9792f3a0ec2ae3cf67d256.jpg',
    id: 'jao89jaoim2',
  },
  {
    userId: '12345',
    date: {
      years: 1,
      months: 7,
    },
    content: '2 Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
    image: 'https://s-media-cache-ak0.pinimg.com/564x/a0/71/6c/a0716ce6fc9792f3a0ec2ae3cf67d256.jpg',
    id: 'oqjeoqjdiwejco8qw',
  },

];

export default firsts;
