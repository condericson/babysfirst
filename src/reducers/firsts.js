function postFirst(state = [], action) {
  switch(action.type){
    case 'ADD_FIRST':
    // return the new state with the new comment
      return [...state, {
        userId: action.userId,
        date: action.date,
        content: action.content
      }];
    default:
      return state;
  }
}



export default postFirst;
