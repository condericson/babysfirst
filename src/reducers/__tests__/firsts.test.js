import firstsReducer from '../firsts';
import { getFirstsSuccess, loadMoreSuccess, addFirstsSuccess, deleteFirstsSuccess } from '../../actions/firsts';

const dummyData = {
  firsts1: [{
    _id: 12345,
    date: '01-01-1999',
    content: 'Lorem whatever',
    image: 'baby.jpg',
    userId: '123abc',
  }],
  firsts2: [{
    date: '12-12-2000',
    content: 'Lorem etc',
    image: 'babys.jpg',
    userId: '123abc',
  }],
  firsts3: [{
    date: '11-10-2001',
    content: 'Lorem other lorem',
    image: 'child.jpg',
    userId: '123abc',
  }],
  loading: false,
  errorMessage: false,
};

describe('firsts actions', () => {
  it('#getFirstsSuccess(): should get a lists of firsts', () => {
    expect(
        firstsReducer(
          undefined,
          getFirstsSuccess(
            dummyData.firsts1,
          )
        )
      ).toMatchSnapshot();
  });
  it('#loadMoreSuccess(): should get another list of firsts', () => {
    const state = firstsReducer(
        undefined,
        getFirstsSuccess(
          dummyData.firsts1,
        )
      );
    expect(state).toMatchSnapshot();
    expect(firstsReducer(
        state,
        loadMoreSuccess(
          dummyData.firsts2,
        )
      )).toMatchSnapshot();
  });

  it('#addFirstsSuccess(): should add a first to list of firsts', () => {
    const state = firstsReducer(
    undefined,
      getFirstsSuccess(
        dummyData.firsts1,
      ),
    );
    expect(state).toMatchSnapshot();
    expect(firstsReducer(
      state,
        addFirstsSuccess(
        dummyData.firsts3,
      )
    )).toMatchSnapshot();
  });

  it('#deleteFirstsSuccess(): should add a first to list of firsts', () => {
    const state = firstsReducer(
      undefined,
      getFirstsSuccess(
        dummyData.firsts1,
      ),
    );
    expect(state).toMatchSnapshot();
    expect(firstsReducer(
    state,
      deleteFirstsSuccess(
        dummyData.firsts1,
      )
    )).toMatchSnapshot();
  });
});
