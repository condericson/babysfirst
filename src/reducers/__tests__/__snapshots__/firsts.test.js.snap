exports[`firsts actions #addFirstsSuccess(): should add a first to list of firsts 1`] = `
Object {
  "errorMessage": false,
  "firsts": Array [
    Object {
      "_id": 12345,
      "content": "Lorem whatever",
      "date": "01-01-1999",
      "image": "baby.jpg",
      "userId": "123abc",
    },
  ],
  "loading": false,
  "noMore": false,
}
`;

exports[`firsts actions #addFirstsSuccess(): should add a first to list of firsts 2`] = `
Object {
  "errorMessage": false,
  "firsts": Array [
    Object {
      "content": "Lorem other lorem",
      "date": "11-10-2001",
      "image": "child.jpg",
      "userId": "123abc",
    },
  ],
  "loading": false,
  "noMore": false,
}
`;

exports[`firsts actions #deleteFirstsSuccess(): should add a first to list of firsts 1`] = `
Object {
  "errorMessage": false,
  "firsts": Array [
    Object {
      "_id": 12345,
      "content": "Lorem whatever",
      "date": "01-01-1999",
      "image": "baby.jpg",
      "userId": "123abc",
    },
  ],
  "loading": false,
  "noMore": false,
}
`;

exports[`firsts actions #deleteFirstsSuccess(): should add a first to list of firsts 2`] = `
Object {
  "errorMessage": false,
  "firsts": Array [
    Object {
      "_id": 12345,
      "content": "Lorem whatever",
      "date": "01-01-1999",
      "image": "baby.jpg",
      "userId": "123abc",
    },
  ],
  "loading": false,
  "noMore": false,
}
`;

exports[`firsts actions #getFirstsSuccess(): should get a lists of firsts 1`] = `
Object {
  "errorMessage": false,
  "firsts": Array [
    Object {
      "_id": 12345,
      "content": "Lorem whatever",
      "date": "01-01-1999",
      "image": "baby.jpg",
      "userId": "123abc",
    },
  ],
  "loading": false,
  "noMore": false,
}
`;

exports[`firsts actions #loadMoreSuccess(): should get another list of firsts 1`] = `
Object {
  "errorMessage": false,
  "firsts": Array [
    Object {
      "_id": 12345,
      "content": "Lorem whatever",
      "date": "01-01-1999",
      "image": "baby.jpg",
      "userId": "123abc",
    },
  ],
  "loading": false,
  "noMore": false,
}
`;

exports[`firsts actions #loadMoreSuccess(): should get another list of firsts 2`] = `
Object {
  "errorMessage": false,
  "firsts": Array [
    Object {
      "_id": 12345,
      "content": "Lorem whatever",
      "date": "01-01-1999",
      "image": "baby.jpg",
      "userId": "123abc",
    },
    Object {
      "content": "Lorem etc",
      "date": "12-12-2000",
      "image": "babys.jpg",
      "userId": "123abc",
    },
  ],
  "loading": false,
  "noMore": false,
}
`;
